fuse/src/main/java/tachyon/fuse/TachyonFuseFs.java

```diff
@@ -0,0 +1,577 @@
+
+  @Override
+  public int create(String path, @mode_t long mode, FuseFileInfo fi) {
+    int ret = 0;
+    final TachyonURI turi = mPathResolverCache.getUnchecked(path);
+    final int flags = fi.flags.get();
+    LOG.trace("create({}, {}) [Tachyon: {}]", path, Integer.toHexString(flags), turi);
+    LOG.warn("{}: mode is ignored in tachyon-fuse", path);
+    if ((flags & 3) != O_WRONLY.intValue()) {
+      LOG.error("Files can only be created in O_WRONLY mode ({})", path);
+      return -ErrorCodes.EACCES();
+    }
+
+    try {
+
+      final OpenFileEntry ofe = new OpenFileEntry();
+      ofe.mIn = null;
+      ofe.mOut = mTFS.getOutStream(turi);
+      LOG.debug("Tachyon OutStream created for {}", path);
+
+      synchronized (mOpenFilesLock) {
+        if (mOpenFiles.size() == MAX_OPEN_FILES) {
+          LOG.error("Cannot open {}: too many open files", turi);
+          return ErrorCodes.EMFILE();
+        }
+        mOpenFiles.put(mOpenFileIds, ofe);
+        fi.fh.set(mOpenFileIds);
+
+        // Assuming I will never wrap around (2^64 open files are quite a lot anyway)
+        mOpenFileIds += 1;
+      }
+      LOG.debug("{} created and opened in O_WRONLY mode", path);
+
+    } catch (FileAlreadyExistsException e) {
+      LOG.debug("File {} already exists", turi, e);
+      ret = ErrorCodes.EEXIST();
+    } catch (IOException e) {
+      LOG.error("IOException on {}", path, e);
+      ret = ErrorCodes.EFAULT();
+    } catch (TachyonException e) {
+      LOG.error("TachyonException on {}", path, e);
+      ret = ErrorCodes.EFAULT();
+    } catch (Throwable e) {
+      LOG.error("Unexpected exception on {}", path, e);
+      ret = ErrorCodes.EFAULT();
+    }
+
+    return -ret;
+  }
+
+  @Override
+  public int flush(String path, FuseFileInfo fi) {
+    LOG.trace("flush({})", path);
+    final long fd = fi.fh.get();
+    OpenFileEntry oe = null;
+    synchronized (mOpenFilesLock) {
+      oe = mOpenFiles.get(fd);
+    }
+    if (oe == null) {
+      LOG.error("Cannot find fd for {} in table", path);
+      return -ErrorCodes.EBADFD();
+    }
+    if (oe.mOut == null) {
+      LOG.error("{} was not open for writing", path);
+      return -ErrorCodes.EBADFD();
+    }
+    try {
+      oe.mOut.flush();
+    } catch (IOException e) {
+      return -ErrorCodes.EIO();
+    }
+
+    return 0;
+  }
+
+  @Override
+  public int getattr(String path, FileStat stat) {
+    int ret = 0;
+    final TachyonURI turi = mPathResolverCache.getUnchecked(path);
+    LOG.trace("getattr({}) [Tachyon: {}]", path, turi);
+    try {
+      final TachyonFile tf = mTFS.openIfExists(turi);
+      if (tf == null) {
+        ret = ErrorCodes.ENOENT();
+      } else {
+        final FileInfo fi = mTFS.getInfo(tf);
+        stat.st_size.set(fi.getLength());
+
+        final long ctime = fi.getLastModificationTimeMs();
+        final long ctime_sec = fi.getLastModificationTimeMs() / 1000;
+        final long ctime_nsec = (fi.getLastModificationTimeMs() % 1000) * 1000;
```

Comment:
ctime_nsec contains the "residual" fraction of seconds not reflected in ctime_sec